### ISSUE TITLE
A CR update caused nil pointer dereference in kube-apiserver 1.9.8 and 1.7.15

### ISSUE THREAD

**Issue by aanm** [kubernetes/kubernetes#67245]
/kind bug

Unfortunately we can't reproduce it but the stack trace might help figure out where the issue is.

In k8s 1.9.8
```
Trace[418895602]: [1m0.000482196s] [1m0.000482196s] END
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0xf2dcf2]

goroutine 30886076 [running]:
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.AuditEventFrom(0x0, 0x0, 0xc4255fc740)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request/context.go:158 +0x22
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObjectNegotiated(0x0, 0x0, 0x8e5eea0, 0xc4255fc740, 0xc42532ea70, 0x9, 0xc42532ea7a, 0x2, 0x7f4cd07b81b0, 0xc423bc0f00, ...)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:118 +0x1b6
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.ErrorNegotiated(0x0, 0x0, 0x8e3ad60, 0xc4258ad460, 0x8e5eea0, 0xc4255fc740, 0xc42532ea70, 0x9, 0xc42532ea7a, 0x2, ...)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:142 +0x163
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.(*RequestScope).err(0xc4249d4f00, 0x8e3ad60, 0xc4258ad460, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:67 +0x10c
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.UpdateResource.func1(0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/update.go:55 +0x1acd
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver.(*crdHandler).ServeHTTP(0xc420448b00, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go:211 +0xcd1
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc422a09e00, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:248 +0x26d
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc42013be30, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0xa1
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3ae4107, 0x17, 0xc42082c870, 0xc42013be30, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x6ad
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc420969b00, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	<autogenerated>:1 +0x75
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc4264d5000, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:254 +0x407
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc42063e2a0, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0xa1
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3ac5cb4, 0xe, 0xc42082ccf0, 0xc42063e2a0, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x6ad
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc4208a97e0, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	<autogenerated>:1 +0x75
k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver.(*proxyHandler).ServeHTTP(0xc424860180, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go:91 +0x15f
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc428d8ab40, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:248 +0x26d
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc421985570, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0xa1
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3ac89a1, 0xf, 0xc421a5d9e0, 0xc421985570, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x6ad
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc423abf580, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	<autogenerated>:1 +0x75
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:51 +0x37d
net/http.HandlerFunc.ServeHTTP(0xc4229ea320, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:165 +0x42a
net/http.HandlerFunc.ServeHTTP(0xc422a32f00, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:49 +0x203a
net/http.HandlerFunc.ServeHTTP(0xc4229ea370, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:79 +0x2b1
net/http.HandlerFunc.ServeHTTP(0xc4229ea3c0, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.WithRequestContext.func1(0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request/requestcontext.go:110 +0xcb
net/http.HandlerFunc.ServeHTTP(0xc423abf5a0, 0x7f4cd07b81b0, 0xc423bc0f00, 0xc425f33200)
	/usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc423abf640, 0x8e60ae0, 0xc423bc0f00, 0xc425f33200, 0xc42b5e80c0)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:93 +0x8d
created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:92 +0x1ab
```

In k8s 1.7.15
```
2018-08-08T16:43:10.155770244Z 	panic: runtime error: invalid memory address or nil pointer dereference
2018-08-08T16:43:10.155778545Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x13080af]
2018-08-08T16:43:10.15578665Z 
2018-08-08T16:43:10.155794266Z goroutine 283370 [running]:
2018-08-08T16:43:10.155802138Z k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0xc42fb7bf18, 0x1, 0x1)
2018-08-08T16:43:10.155810691Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
2018-08-08T16:43:10.155820207Z panic(0x35e92c0, 0x798c590)
2018-08-08T16:43:10.155830294Z 	/usr/local/go/src/runtime/panic.go:489 +0x2cf
2018-08-08T16:43:10.155839316Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.AuditEventFrom(0x0, 0x0, 0xc421bb6360)
2018-08-08T16:43:10.155848311Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request/context.go:158 +0x2f
2018-08-08T16:43:10.155857251Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObjectNegotiated(0x0, 0x0, 0x7809f20, 0xc421bb6360, 0xc4298dc160, 0x9, 0xc4298dc16a, 0x2, 0x78022e0, 0xc4208a9650, ...)
2018-08-08T16:43:10.155866794Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:108 +0xab
2018-08-08T16:43:10.155876643Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.ErrorNegotiated(0x0, 0x0, 0x77ddfe0, 0xc4217f0380, 0x7809f20, 0xc421bb6360, 0xc4298dc160, 0x9, 0xc4298dc16a, 0x2, ...)
2018-08-08T16:43:10.15588616Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:135 +0x165
2018-08-08T16:43:10.155895782Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.(*RequestScope).err(0xc424d2c500, 0x77ddfe0, 0xc4217f0380, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.155905237Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:80 +0x10e
2018-08-08T16:43:10.155925835Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.UpdateResource.func1(0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.155935594Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:916 +0xfb2
2018-08-08T16:43:10.155945467Z k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver.(*crdHandler).ServeHTTP(0xc4203a4880, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156130049Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go:188 +0xe25
2018-08-08T16:43:10.156148672Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc4203d91c0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.15615763Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:248 +0x3dd
2018-08-08T16:43:10.156166819Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc4200488c0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156175652Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x72
2018-08-08T16:43:10.156185829Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3c6843f, 0x17, 0xc4201765a0, 0xc4200488c0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156213522Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x301
2018-08-08T16:43:10.156223679Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc420329240, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156232638Z 	<autogenerated>:64 +0x86
2018-08-08T16:43:10.156241218Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc4217e5740, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156250132Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:254 +0x26e
2018-08-08T16:43:10.156259844Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc4201084d0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156269058Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x72
2018-08-08T16:43:10.156278418Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3c44120, 0xe, 0xc4207ea2d0, 0xc4201084d0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156291109Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x301
2018-08-08T16:43:10.156315312Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc4203a1300, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156325362Z 	<autogenerated>:64 +0x86
2018-08-08T16:43:10.156333856Z k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver.(*proxyHandler).ServeHTTP(0xc421820700, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156342634Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go:91 +0x122
2018-08-08T16:43:10.156352099Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc42213c580, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156360875Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:248 +0x3dd
2018-08-08T16:43:10.156369615Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc4287f00e0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156378272Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x72
2018-08-08T16:43:10.15638734Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3c47727, 0xf, 0xc42c37c900, 0xc4287f00e0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156396239Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:161 +0x301
2018-08-08T16:43:10.156405244Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc422d830c0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156413416Z 	<autogenerated>:64 +0x86
2018-08-08T16:43:10.156421089Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156429687Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:50 +0x30c
2018-08-08T16:43:10.156438502Z net/http.HandlerFunc.ServeHTTP(0xc421d41a40, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156446335Z 	/usr/local/go/src/net/http/server.go:1942 +0x44
2018-08-08T16:43:10.156454212Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156462739Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:47 +0x226a
2018-08-08T16:43:10.156471182Z net/http.HandlerFunc.ServeHTTP(0xc421d41c00, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.15647883Z 	/usr/local/go/src/net/http/server.go:1942 +0x44
2018-08-08T16:43:10.156487362Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156506868Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:73 +0x2b0
2018-08-08T16:43:10.156517012Z net/http.HandlerFunc.ServeHTTP(0xc4284bc050, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156525607Z 	/usr/local/go/src/net/http/server.go:1942 +0x44
2018-08-08T16:43:10.156533586Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.WithRequestContext.func1(0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156541869Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request/requestcontext.go:110 +0xef
2018-08-08T16:43:10.156550912Z net/http.HandlerFunc.ServeHTTP(0xc422d830e0, 0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156558949Z 	/usr/local/go/src/net/http/server.go:1942 +0x44
2018-08-08T16:43:10.156567373Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithPanicRecovery.func1(0x78022e0, 0xc4208a9650, 0xc42124b900)
2018-08-08T16:43:10.156576038Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/wrap.go:41 +0x11b
2018-08-08T16:43:10.156584626Z net/http.HandlerFunc.ServeHTTP(0xc422d83180, 0x7ffff7999580, 0xc426ebc500, 0xc42124b900)
2018-08-08T16:43:10.156592901Z 	/usr/local/go/src/net/http/server.go:1942 +0x44
2018-08-08T16:43:10.15660329Z k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc422d831a0, 0x780bb20, 0xc426ebc500, 0xc42124b900, 0xc4223cd560)
2018-08-08T16:43:10.156618317Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:89 +0x8d
2018-08-08T16:43:10.156628076Z created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
2018-08-08T16:43:10.156636857Z 	/workspace/anago-v1.7.15-beta.0.16+8c7c1c8b0ce086/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:91 +0x1c0
```

**Comment by nikhita** [comment]
@aanm this looks more like a custom resource update than a CRD update. :eyes: 

Can you clarify if it was a CRD or a CR?

**Comment by nikhita** [comment]
In any case, it appears to be originating from here:

> https://github.com/kubernetes/kubernetes/blob/c138b85178156011dc934c2c9f4837476876fb07/staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go#L92-L93

Maybe there is a case when the `timeoutHandler.handler` is nil...?

**Comment by nikhita** [comment]
> k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.AuditEventFrom(0x0, 0x0, 0xc4255fc740)
	/workspace/anago-v1.9.8-beta.0.40+c138b851781560/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request/context.go:158 +0x22

https://github.com/kubernetes/kubernetes/blob/c138b85178156011dc934c2c9f4837476876fb07/staging/src/k8s.io/apiserver/pkg/endpoints/request/context.go#L157-L160

**Comment by aanm** [comment]
@nikhita Sorry, I meant the CR not CRD.

**Comment by sttts** [comment]
I believe this got fixed in 1.10 with https://github.com/kubernetes/kubernetes/pull/60628.

**Comment by liggitt** [comment]
> I believe this got fixed in 1.10 with #60628.

agreed.

/close
